### PR TITLE
fix(ecs): setting updatePolicy results in error due to updateType having default value 

### PR DESCRIPTION
**Pull request by richarddotcodes**
Currently, providing an updatePolicy always results in an error
since updateType is always set. This change fixes that.

closes https://github.com/aws/aws-cdk/issues/11581

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -338,7 +338,7 @@ export class Cluster extends Resource implements ICluster {
     const autoScalingGroup = new autoscaling.AutoScalingGroup(this, id, {
       vpc: this.vpc,
       machineImage,
-      updateType: options.updateType || autoscaling.UpdateType.REPLACING_UPDATE,
+      updateType: !!options.updatePolicy ? undefined : options.updateType || autoscaling.UpdateType.REPLACING_UPDATE,
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ecs/lib/cluster.ts:341]: This behavior change should also be reflected in the `@default` docstring of the `updateType` field.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
PRs with changes that are not statically guaranteed by the type system must have tests to make sure that (a) the new behavior works as intended and (b) it is not accidentally undone by future changes (See [Contributing Guide, Work Your Magic](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-3-work-your-magic)).

File: packages/@aws-cdk/aws-ecs/lib/cluster.ts (hunk 1 above)
Comment (richarddotcodes) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts:341]: Is this really changing the default for the field, or am I just changing what gets passed in?

Comment (rix0rrr) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts:341]:
```
`updatePolicy` takes precedence over `updateType`.

@default UpdateType.REPLACING_UPDATE, unless updatePolicy has been set
```

Comment (richarddotcodes) [reply packages/@aws-cdk/aws-ecs/lib/cluster.ts:341]: Thanks. I've updated this PR to reflect the requested changes.

**Comment by richarddotcodes** [comment]
@rix0rrr Can I get a review? Thanks!

**Review by TheRealAmazonKendra — APPROVED** [review]
Looks good!

**Comment by TheRealAmazonKendra** [comment]
@richarddotcodes  In order for Mergify to merge in your change you'll need to [allow changes to your branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

**Comment by richarddotcodes** [comment]
@TheRealAmazonKendra Thanks for the info. I've followed the instructions in the link you'd sent. I also merged in changes from main; can I have another review?

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
https://github.com/Mergifyio update

**Pull request merged**